src/App.tsx

```diff
@@ -238,24 +239,28 @@ async function doChat(
     }
 
     if (toolCallStreamStore.getSnapShot().length > 0) {
-      // do the tool calls
-      await callTools(
+      // Execute the tool calls and check if evm-balances suppresses response
+      const nextShouldSuppressResponse = await callTools(
         toolCallStreamStore,
         messageHistoryStore,
         executeOperation,
+        shouldSuppressResponse,
       );
 
-      // inform the model of the tool call responses
-      await doChat(
-        controller,
-        client,
-        messageHistoryStore,
-        tools,
-        progressStore,
-        messageStore,
-        toolCallStreamStore,
-        executeOperation,
-      );
+      // Allow chat to continue only if no evm-balances suppression OR a reset occurred
+      if (!nextShouldSuppressResponse) {
```

Comment:
ideal when the model requests a tool call, we do give it the response, not sure how this would affect subsequent tool calls or if it's okay to not have a response when a model requests a tool call.

Reply:
Open AI's schema is to summarize the results of a tool call with natural language, so we are kinda implementing an anti-pattern, but I think it makes sense for this use case.

We are still getting the result of the tool call, like we would for cancelling a tx in metamask.


![Screenshot 2025-01-29 at 3 48 39 PM](https://github.com/user-attachments/assets/add71c86-cc0c-45db-82b0-64de1b65aca3)

Reply:
It's kinda like we are adding another base case to exit the recursion.